amethyst_tiles/src/lib.rs

```diff
@@ -26,7 +26,7 @@ use amethyst_core::math::Vector3;
 pub trait CoordinateEncoder: 'static + Clone + Default + Send + Sync {
     /// Constructor interface for `Self` which consumes the maps dimensions, which is required for some
     /// encoding types to fit within a given coordinate space.
-    fn from_dimensions(x: u32, y: u32, z: u32) -> Self;
+    fn from_dimensions(dimensions: &Vector3<u32>) -> Self;
```

Comment:
Should we just pass `Vector3<u32>` by value? It's [`Copy`](https://nalgebra.org/rustdoc/nalgebra/base/struct.Matrix.html#impl-Copy)